pandas/io/formats/csvs.py

```diff
@@ -307,7 +307,10 @@ def _save_chunk(self, start_i: int, end_i: int) -> None:
         df = self.obj.iloc[slicer]
 
         res = df._mgr.to_native_types(**self._number_format)
-        data = [res.iget_values(i) for i in range(len(res.items))]
+        data = [
+            self._change_to_str_date_format(res.iget_values(i))
+            for i in range(len(res.items))
```

Comment:
This is the wrong place for this. We have formatters per dtype, it would have to be added there

Reply:
Gotcha. Thanks for pointing it out. Very helpful.